benchcab/build_cable.py

```diff
@@ -146,26 +140,46 @@ def clean_if_needed():
             raise ("Error cleaning previous compilation")
 
 
-def build_cable_offline(branch_name: str, modules: list):
-    build_dir = Path(CWD, SRC_DIR, branch_name, "offline")
-    os.chdir(build_dir)
+def prepare_build(
+    build_script: str, branch_name: str, modules: list, root_dir=CWD
+) -> Path:
+    """Prepare the build script depending if a build script was specified
+    in the configuration or not."""
 
-    clean_if_needed()
-    ofname = adjust_build_script(modules)
+    if build_script:
+        build_script_path = Path(root_dir, SRC_DIR, branch_name, build_script)
+        assert (
+            build_script_path.is_file(),
+            f"The build script specified in the config.yaml file, {build_script_path}, "
+            "is not a valid file.",
+        )
 
-    cmd = "chmod +x %s" % (ofname)
-    error = subprocess.call(cmd, shell=True)
-    if error == 1:
-        raise ("Error changing file to executable")
+    else:
+        default_script_path = Path(
+            root_dir, SRC_DIR, branch_name, "offline", "build3.sh"
+        )
+        assert (
+            default_script_path.is_file(),
+            f"The default build script, {default_script_path}, could not be found."
+            "Do you need to specify a different build script with the "
+            "'build_script' option in config.yaml?",
+        )
```

Comment:
We should probably raise a `RuntimeError` exception instead of assert because assert can get optimised out